pilot/pkg/bootstrap/server.go

```diff
@@ -1307,14 +1317,13 @@ func (s *Server) initWorkloadTrustBundle(args *PilotArgs) error {
 	return nil
 }
 
-// isCADisabled returns whether CA functionality is disabled in istiod.
-// It returns true only if istiod certs is signed by Kubernetes or
-// workload certs are signed by external CA
-func (s *Server) isCADisabled() bool {
+// isK8SSigning returns whether K8S is used to sign certs instead of private keys known by Istiod
+func (s *Server) isK8SSigning() bool {
 	if s.RA == nil {
 		return false
 	}
 	// do not create CA server if PilotCertProvider is `kubernetes` and RA server exists
+	// TODO(costin): this is dead code
```

Comment:
What do you mean by dead code?

Reply:
Chiron code now refuses to sign with 'direct k8s' - so if user sets this option, cert signing will not work and the setup is broken. 

Unless I missed some code path - which is likely...  But I think K8S itself will reject this now, so dead either way.

Reply:
Not that. When it is  CertProviderKubernetes, istio create a CSR, can approve it. It does not check the signer at all, and donot care who signs it.

Reply:
Right - but https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers states 

    "The stable CertificateSigningRequest API (version certificates.k8s.io/v1 and later) does not allow to set the signerName as kubernetes.io/legacy-unknown."

We are already blocking legacy-unknown from creating Istiod DNS certs - I don't know how you would start Istiod, perhaps with the custom TLS certs - but in any cluster that has 'v1' resource the cert signing will fail anyways.

I don't know what is the oldest version or if any K8S still in support window has the old CertSigner version - so perhaps it's 'zombie' code.

Reply:
To clarify: CertProviderKubernetes sends "" as signer name. I did not find any code to set "kubernetes.io/legacy-unknown" for this case - so I really have no idea how it can work unless you have a VERY old cluster. 

I'm not removing this - I can add more explanations to the comment, but legacy-unknown needs to be cleaned up eventually.

Reply:
#50270 removed a lot around it, so anything left is really dead code